dcore/configuration.py

```diff
@@ -0,0 +1,81 @@
+"""
+Some simple tools for making model configuration nicer.
+"""
+
+
+class Configuration(object):
+    def __init__(self, **kwargs):
+
+        self.__setattr__ = self._proto__setattr__
+        for name, value in kwargs.iteritems():
+            print name, value
```

Comment:
Hm, I hadn't really thought about parallel - but it is useful to have a record of which parameters are different from the default. I do tend to pipe the output to a log file but maybe there's a better way of doing this?


________________________________
From: Lawrence Mitchell <notifications@github.com>
Sent: 09 March 2016 19:09
To: firedrakeproject/dcore
Cc: Shipton, Jemma
Subject: Re: [dcore] Tidy configuration (#20)


In dcore/configuration.py<https://github.com/firedrakeproject/dcore/pull/20#discussion_r55570803>:

> @@ -0,0 +1,81 @@
> +"""
> +Some simple tools for making model configuration nicer.
> +"""
> +
> +
> +class Configuration(object):
> +    def __init__(self, **kwargs):
> +
> +        self.__setattr__ = self._proto__setattr__
> +        for name, value in kwargs.iteritems():
> +            print name, value


Do you really mean to print the values when initialising the configuration object. This is pretty unfriendly in parallel.

-
Reply to this email directly or view it on GitHub<https://github.com/firedrakeproject/dcore/pull/20/files#r55570803>.